dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -880,6 +881,8 @@ private Class<? extends Workflow> getSubClass(WorkflowSubClass subclass) {
             targetClass = BioWorkflow.class;
         } else if (subclass == WorkflowSubClass.APPTOOL) {
             targetClass = AppTool.class;
+        } else if (subclass == WorkflowSubClass.NOTEBOOK) {
```

Comment:
You're following the existing pattern, so you can leave it, but a suggestion would be to move this logic to the enum, and then if we add additional subclasses, we don't have to update this code every time, or worse, risk overlooking it. Something like, untested, in WorkflowSubClass.java:

```
public enum WorkflowSubClass {
    BIOWORKFLOW(BioWorkflow.class),
    SERVICE(Service.class),
    APPTOOL(AppTool.class),
    NOTEBOOK(Notebook.class);

    private final Class<? extends Workflow> targetClass;
    
    WorkflowSubClass(Class<? extends Workflow> targetClass) {
        this.targetClass = targetClass;
    }

    public Class<? extends Workflow> getTargetClass() {
        return targetClass;
    }
}
```

Then this whole method just becomes `return subclass.getTargetClass()`, unless you also need to do a null check. And if you don't need the null check, you can probably just get rid of the method.